# 07 EDA — Sellers

Read-only inspection of `olist_sellers_dataset.csv`. No cleaning, charts, or feature engineering.

## 1. Import libraries

Import pandas for tabular inspection.

In [1]:
import pandas as pd

**Observation:** pandas is ready.

## 2. Load CSV

Load only this CSV into memory without changing the source.

In [2]:
file_path = '../data/raw/olist/olist_sellers_dataset.csv'
df = pd.read_csv(file_path)

**Observation:** The table is loaded; the raw CSV is unchanged.

## 3. Shape

Check table size.

In [3]:
df.shape

(3095, 4)

**Observation:** This reports the table's row and column count.

## 4. First 5 rows

Preview the raw structure.

In [4]:
df.head()

                          seller_id  ...  seller_state
0  3442f8959a84dea7ee197c632cb2df15  ...            SP
1  d1b65fc7debc3361ea86b5f14c68d2e2  ...            SP
2  ce3ad9de960102d0677a81f5d0bb7b2d  ...            RJ
3  c0f3eea2e14555b6faeea3dd58c1b1c3  ...            SP
4  51a04a8a6bdcb23deccc82b0b80742cf  ...            SP

[5 rows x 4 columns]

**Observation:** The preview confirms the row grain and field layout.

## 5. Column names

List available fields.

In [5]:
df.columns.tolist()

['seller_id', 'seller_zip_code_prefix', 'seller_city', 'seller_state']

**Observation:** These are the fields available for this table's analysis.

## 6. Data types

Review pandas data types.

In [6]:
df.dtypes

seller_id                   str
seller_zip_code_prefix    int64
seller_city                 str
seller_state                str
dtype: object

**Observation:** Types show which fields are text, numeric, or converted datetime values.

## 7. Missing values

Count nulls by field.

In [7]:
df.isna().sum()

seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

**Observation:** Missing counts identify incomplete fields without changing them.

## 8. Missing percentage

Measure missingness relative to table size.

In [8]:
(df.isna().mean().mul(100).round(2)).sort_values(ascending=False)

seller_id                 0.0
seller_zip_code_prefix    0.0
seller_city               0.0
seller_state              0.0
dtype: float64

**Observation:** Percentages show the relative scale of incomplete fields.

## 9. Duplicate rows

Count complete row duplicates.

In [9]:
df.duplicated().sum()

np.int64(0)

**Observation:** This is the number of repeated full rows; none are removed.

## 10. Duplicate primary keys

Validate the documented candidate key.

In [10]:
primary_key = ['seller_id']
df.duplicated(subset=primary_key, keep=False).sum()

np.int64(0)

**Observation:** This counts rows involved in duplicate candidate keys for seller_id.

## 11. Unique values

Inspect meaningful category or entity counts.

In [11]:
pd.Series({'sellers': df['seller_id'].nunique(), 'states': df['seller_state'].nunique(), 'cities': df['seller_city'].nunique(), 'zip_prefixes': df['seller_zip_code_prefix'].nunique()})

sellers         3095
states            23
cities           611
zip_prefixes    2246
dtype: int64

**Observation:** This gives a concise view of important distinct values.

## 12. Basic describe()

Generate a basic summary for all fields.

In [12]:
df.describe(include='all')

                               seller_id  ...  seller_state
count                               3095  ...          3095
unique                              3095  ...            23
top     3442f8959a84dea7ee197c632cb2df15  ...            SP
freq                                   1  ...          1849
mean                                 NaN  ...           NaN
std                                  NaN  ...           NaN
min                                  NaN  ...           NaN
25%                                  NaN  ...           NaN
50%                                  NaN  ...           NaN
75%                                  NaN  ...           NaN
max                                  NaN  ...           NaN

[11 rows x 4 columns]

**Observation:** The summary describes completeness, distinctness, common values, and numeric ranges where applicable.

## 13. Table-specific observations

The table identifies 3,095 marketplace sellers and their location, supporting future seller and geographic comparisons.

## 14. Data quality observations

No missing or exact duplicate rows were found, and `seller_id` is valid. Seven seller ZIP prefixes have no geolocation match.